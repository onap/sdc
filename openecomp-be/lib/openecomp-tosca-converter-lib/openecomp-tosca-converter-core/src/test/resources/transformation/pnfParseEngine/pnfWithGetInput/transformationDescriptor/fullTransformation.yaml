- transformation-for: nodeTemplate
  name: PnfExtCp transformation
  description: converts any node template that has a type tosca.nodes.nfv.PnfExtCp
  query:
    type: tosca.nodes.nfv.PnfExtCp
  conversions:
    - query:
        properties:
          layer_protocols:
      to-name: ip_requirements
      to-value:
        strategy: replaceInList
        list:
          - from: ipv4
            to:
              assingment_method: dhcp
              ip_version: 4
          - from: ipv6
            to:
              assingment_method: dhcp
              ip_version: 6
      to-get-input: ipRequirements
    - query:
        properties:
          role:
      to-name: role
      to-value:
        strategy: copy
      to-get-input: stringTransformation
- transformation-for: getInputFunction
  name: ipRequirements
  description: converts IpRequirements inputs when called by a get_input TOSCA function
  conversions:
    - query:
        type:
      to-name: type
      to-value:
        strategy: copy
    - query:
        description:
      to-name: description
      to-value:
        strategy: copy
    - query:
        entry_schema:
          type:
      to-name: type
      to-value:
        strategy: replace
        from: string
        to: org.openecomp.datatypes.network.IpRequirements
    - query:
        default:
      to-name: default
      to-value:
        strategy: replaceInList
        list:
          - from: ipv4
            to:
              assingment_method: dhcp
              ip_version: 4
          - from: ipv6
            to:
              assingment_method: dhcp
              ip_version: 6
- transformation-for: getInputFunction
  name: stringTransformation
  description: converts a string input
  conversions:
    - query:
        type:
      to-name: type
      to-value:
        strategy: copy
    - query:
        description:
      to-name: description
      to-value:
        strategy: copy
    - query:
        default:
      to-name: default
      to-value:
        strategy: copy