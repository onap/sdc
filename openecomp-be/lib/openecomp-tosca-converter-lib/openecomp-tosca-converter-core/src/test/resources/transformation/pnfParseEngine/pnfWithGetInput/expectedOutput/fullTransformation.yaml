topology_template:
  inputs:
    role1:
      type: string
      description: Role
      default: leaf
    layer_protocols:
      type: list
      description: IP protocols
      entry_schema:
        type: org.openecomp.datatypes.network.IpRequirements
      default:
        - assingment_method: dhcp
          ip_version: 4
        - assingment_method: dhcp
          ip_version: 6
    role:
      type: string
      description: Role
      default: leaf
  node_templates:
    pnfExtCp_3:
      properties:
        role: {get_input: role1}
    pnfExtCp_2:
      properties:
        ip_requirements: {get_input: layer_protocols}
        role: {get_input: role}
    pnfExtCp_1:
      properties:
        ip_requirements:
          - assingment_method: dhcp
            ip_version: 4
          - assingment_method: dhcp
            ip_version: 6
        role: leaf